script-test/playbackstrategies/modifiers/html5tests.js

```diff
@@ -0,0 +1,3260 @@
+              expect(player.getState()).toEqual(MediaPlayerBase.STATE.ERROR);
+            });
+
+            it('Calling Begin Playback From In Error State Is An Error', function () {
+              recentEvents = [];
+              player.beginPlaybackFrom();
+
+              expect(logger.error).toHaveBeenCalledWith('Cannot beginPlaybackFrom while in the \'ERROR\' state');
+              expect(player.getState()).toEqual(MediaPlayerBase.STATE.ERROR);
+            });
+
+            it('Calling Initialise Media In Error State Is An Error', function () {
+              recentEvents = [];
+              player.initialiseMedia(MediaPlayerBase.TYPE.VIDEO, 'testUrl', 'testMimeType', sourceContainer, {});
+
+              expect(logger.error).toHaveBeenCalledWith('Cannot set source unless in the \'EMPTY\' state');
+              expect(player.getState()).toEqual(MediaPlayerBase.STATE.ERROR);
+            });
+
+            it('Calling Play From In Error State Is An Error', function () {
+              recentEvents = [];
+              player.playFrom();
+
+              expect(logger.error).toHaveBeenCalledWith('Cannot playFrom while in the \'ERROR\' state');
+              expect(player.getState()).toEqual(MediaPlayerBase.STATE.ERROR);
+            });
+
+            it('Calling Pause In Error State Is An Error', function () {
+              recentEvents = [];
+              player.pause();
+
+              expect(logger.error).toHaveBeenCalledWith('Cannot pause while in the \'ERROR\' state');
+              expect(player.getState()).toEqual(MediaPlayerBase.STATE.ERROR);
+            });
+
+            it('Calling Resume In Error State Is An Error', function () {
+              recentEvents = [];
+              player.resume();
+
+              expect(logger.error).toHaveBeenCalledWith('Cannot resume while in the \'ERROR\' state');
+              expect(player.getState()).toEqual(MediaPlayerBase.STATE.ERROR);
+            });
+
+            it('Calling Stop From In Error State Is An Error', function () {
+              recentEvents = [];
+              player.stop();
+
+              expect(logger.error).toHaveBeenCalledWith('Cannot stop while in the \'ERROR\' state');
+              expect(player.getState()).toEqual(MediaPlayerBase.STATE.ERROR);
+            });
+
+            it('Calling Reset In Error State Goes To Empty State', function () {
+              player.reset();
+
+              expect(player.getState()).toEqual(MediaPlayerBase.STATE.EMPTY);
+            });
+
+            it('When Buffering Finishes During Error We Continue To Be In Error', function () {
+              expect(player.getState()).toEqual(MediaPlayerBase.STATE.ERROR);
+
+              metaDataCallback();
+              finishedBufferingCallback();
+
+              expect(player.getState()).toEqual(MediaPlayerBase.STATE.ERROR);
+            });
+          });
+
+          describe('Error logged for invalid state transitions tests', function () {
```

Comment:
Could possibly fold these into other error state tests.

Reply:
These are testing calls on methods when in states that are incorrect, the other is testing functionality within only the error state.